azure-fluid-relay-preview-pr/articles/howtos/hosting.md

```diff
@@ -0,0 +1,245 @@
+
+# Pre-requisites
+1. Add the following extensions to Visual Studio Code: 
+    - [Azure Account](https://marketplace.visualstudio.com/items?itemName=ms-vscode.azure-account): It is required to sign in to your `Azure Subscription`.
+    - [Azure App Service](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azureappservice): It is required for deploying app using `App Services`.
+    - [Azure Static Web App](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurestaticwebapps): It is required for deploying app using `Static Web Apps`.
+
+These extensions allow the developer to use Azure directly from Visual Studio Code.
+
+2. Run the following command to clone the repository:
+
+```shell
+git clone https://github.com/microsoft/FluidHelloWorld
+```
+
+# Connecting to Azure Fluid Relay Service
+You can connect to `Azure Fluid Relay Service` by providing the tenant ID and key that is uniquely generated for you when onboarding to the service. The `tokenProvider` can be generated using any one of the following methods: 
+
+1. `InsecureTokenProvider`: This type of implementation is only recommended for development purposes as it exposes the tenant key secret in your client-side code bundle.
+
+```typescript
+import { AzureClient, AzureConnectionConfig } from "@fluidframework/azure-client";
+
+const config: AzureConnectionConfig = {
+    tenantId: "myTenantId",
+    tokenProvider: new InsecureTokenProvider("myTenantKey", { id: "123", name: "Test User" }),
+    orderer: "https://myOrdererUrl",
+    storage: "https://myStorageUrl",
+};
+const client = new AzureClient(config);
+```
+
+2. `AzureFunctionTokenProvider`: This implementation can be used for fetching the token from your own backend service that is responsible for signing it with the tenant key. It provides you with a secured way to generate token and pass the generated token back to the client app. The `AzureFunctionTokenProvider` is an implementation that fulfills the `ITokenProvider` interface without exposing the tenant key secret in client-side code. Developers would require their own backend `HTTPTrigger Azure Function` for fetching the token, thereby passing the url to `AzureFunctionTokenProvider`. 
+
+```typescript
+import { AzureClient, AzureConnectionConfig } from "@fluidframework/azure-client";
+
+const config: AzureConnectionConfig = {
+    tenantId: "myTenantId",
+    tokenProvider: new AzureFunctionTokenProvider("myAzureFunctionUrl"+"/api/GetFrsToken", { userId: "test-user",userName: "Test User" }),
+    orderer: "https://myOrdererUrl",
+    storage: "https://myStorageUrl",
+};
+const client = new AzureClient(config);
+```
+<br/>
+
+The below code snippet can help you with creating `HTTPTrigger` function for fetching the token by passing in your tenant key.
```

Comment:
Again, I think devs would need some guidance on where to place the code for this snippet. Perhaps consider including it in step 3 below where we discuss adding the `api` folder and the `HTTPTrigger`

Reply:
The main reason behind writing about the trigger and code snippet here was:
1. For App Services, the developer would require a backend Azure Function to fetch the token
2. For Static, there are two options: backend trigger or `api`

I thought it would be better to add the trigger and code snippet in the 2nd point and later on give the `api` option used in `Static Web Apps`.  But I think, the other way round also works well.

Reply:
Makes sense, let's also consider adding explicit instructions here that the devs need to create the `api` folder and then place this code in there. Main concern here is that they won't know where to place this code and have it be accessible to the BE API through the yml file

Reply:
There is an explanation saying:  "Developers can create their own `HTTPTrigger Azure Function` under an `api` folder in the `FluidHelloWorld` repository itself. Below, `GetFrsToken` is an `HTTPTrigger Azure Function` which is responsible for fetching token from your very own backend."

And, an image of how the `api` folder would look with `GetFrsToken` as the `HTTPTrigger` function. 

I can add how to `api` can be accessed through the yml field. The is some explanation for the same in `Deployment using Azure Static Web App`